pkgs/misc/vim-plugins/update.py

```diff
@@ -488,25 +470,49 @@ def parse_args():
         action="store_true",
         help="Automatically commit updates",
     )
+    parser.add_argument(
+        "--allow-dirty",
+        dest="allow_dirty",
+        action="store_true",
+        help=(
+            "Allow commit to continue even if state is unexpectedly dirty. "
+            "This is only helpful when developing vimPlugins infrastructure."
+        ),
+    )
 
     return parser.parse_args()
 
 
-def get_nixpkgs_repo() -> git.Repo:
-    repo = git.Repo(NIXPKGS_PATH)
-    if repo.is_dirty():
-        raise Exception("Please stash your changes before updating.")
-    return repo
+class NixpkgsRepo:
+    def __init__(self, allow_dirty: bool):
+        self.allow_dirty: bool = allow_dirty
```

Comment:
Why do we need to store this? Wouldn't it be easier to just check it right here?

Reply:
In general I think the class is not really necessary here.

Reply:
Yeah, I didn't really think you'd like it and this commit can be reverted if you like. My goal was to keep `main` as a short and readable description of the control flow by moving some state and implementation details out of it. Not a big deal either way.

Reply:
I wonder if you'd prefer a decorator to a class. It would be more functional, achieve the same goal of `main` readability, and fix the semantic problem with `PluginUpdater`. The downside is that decorators themselves aren't necessarily all that readable...

Reply:
Oh sorry, those comments were directed towards your comment on `PluginUpdater` which I've now removed in favor of a closure. This one I agree on.